tests/PerformanceTests/Microsoft.Identity.Web.Perf.Client/TestRunner.cs

```diff
@@ -69,56 +70,92 @@ public async Task Run()
 
             IEnumerable<Task> CreateTasks()
             {
-                // Checks if processing end time has passed
-                yield return Task.Run(
-                    async () =>
-                    {
-                        while (_continueProcessing && DateTime.Now < _processingEndTime)
-                        {
-                            await Task.Delay(30000);
-                        }
+                var tokenSource = new CancellationTokenSource();
 
-                        StopProcessing();
-                        Console.WriteLine("Processing stopped. Duration elapsed.");
-                    });
+                if (!_options.RunIndefinitely)
```

Comment:
Is there any more work before committing this? It feel like it has been open (and approved) for a very long time.